doc/man3/EVP_PKEY_new.pod

```diff
@@ -168,7 +168,19 @@ general private key without reference to any particular algorithm.
 The structure returned by EVP_PKEY_new() is empty. To add a private or public
 key to this empty structure use the appropriate functions described in
 L<EVP_PKEY_set1_RSA(3)>, L<EVP_PKEY_set1_DSA(3)>, L<EVP_PKEY_set1_DH(3)> or
-L<EVP_PKEY_set1_EC_KEY(3)>.
+L<EVP_PKEY_set1_EC_KEY(3)> for legacy key types or for fully provider-managed
+key types (see L<provider-keymgmt(7)>), functions such as
+L<EVP_PKEY_set1_encoded_public_key(3)>.
+
+Generally caution is advised for using an B<EVP_PKEY> structure across
```

Comment:
I am not sure I quite understand what this is saying. I assume you are talking about external providers, since internal openssl providers should be fine.

Reply:
What this tries to say is that there's other functions available to "populate" empty keys if a provider (regardless of internal or external) is used to manage the EVP_PKEY.

Reply:
It's a bit misleading, written like that, as it also seems to say that you could use `EVP_PKEY_set1_encoded_public_key()` instead of `EVP_PKEY_set1_RSA()`...  I suggest making that part a separate sentence, or even paragraph.